### PR TITLE
ci: backport GHA to stable/8.0

### LINKED ISSUE #10009: Backport GitHub CI to stable/8.0 branch
**Description**

In the `main` branch we migrated our CI from Jenkins to GitHub Actions. Right now other branches are still using Jenkins as CI. To align our CI and reduce the maintenance efforts we would like to also use GitHub Actions for our stable branch CI.

This issue is about backporting the GitHub CI to the `stable/8.0` branch.

blocking #9137 and #9138

### PR DESCRIPTION
**Pull request by lenaschoenburg**
This is a manual backup of the github action workflows from `main` to `stable/8.0`. Essentially I copied over the `.github/` folder so some changes to dependabot and issue templates are also included.

The only difference is the removal of the docker checks which are not relevant to 8.0 because the things they are testing are only available on 8.1.

Also backports changes from https://github.com/camunda/zeebe/pull/9462 to ensure that tests work with TCC
Also backports changes from https://github.com/camunda/zeebe/pull/9463 to work around consistent test failures in the atomix module.

closes #10009

### CODE REVIEW

**Comment by oleschoenburg** [comment]
CodeQL warnings can be ignored, they just pop up here because it never ran on `stable/8.0` before.

File: .github/ISSUE_TEMPLATE/bug_report.md [hunk 1]
```diff
@@ -3,7 +3,7 @@
 name: Bug report
 about: Create a report to help us improve
 title: ''
-labels: 'Type: Bug'
```
Comment (Zelldon) [inline .github/ISSUE_TEMPLATE/bug_report.md:6]: :question: Why are these changes necessary ? :thinking:

File: .github/actions/setup-zeebe/action.yml [hunk 2]
```diff
@@ -0,0 +1,61 @@
...
+    description: If true, will set up Golang; defaults to true
+    required: false
+    default: "true"
+  java:
+    description: If true, will set up Java; defaults to true
+    required: false
+    default: "true"
+  java-version:
+    description: The JDK version to setup
+    default: "17"
```
Comment (Zelldon) [inline .github/actions/setup-zeebe/action.yml:19]: :question:  Not sure, but was 8.0 still with JDK 11?

File: .github/workflows/deploy.yml [hunk 3]
```diff
@@ -0,0 +1,122 @@
+name: Deploy
+
+on:
+  push:
+    branches:
+      - main
```
Comment (Zelldon) [inline .github/workflows/deploy.yml:6]: :question: I guess this will not be used then on stable?

File: .github/workflows/qa-testbench.yaml [hunk 4]
```diff
@@ -0,0 +1,172 @@
...
+on:
+  workflow_dispatch:
+    inputs:
+      generation:
+        description: 'Specifies the generation template which should be used by the testbench run'
+        required: false
+        default: 'Zeebe SNAPSHOT'
+      branch:
+        description: 'Specifies the branch, for which the QA Testbench run should be executed'
+        default: 'main'
```
Comment (Zelldon) [inline .github/workflows/qa-testbench.yaml:12]: :question: I guess this should be stable then?

File: .github/workflows/qa-testbench.yaml [hunk 5]
```diff
@@ -0,0 +1,172 @@
...
+    if: ${{ always() }}
+    env:
+      GENERATION_TEMPLATE: "${{ github.event.inputs.generation }}"
+      BRANCH_NAME: "${{ github.event.inputs.branch }}"
+    steps:
+      # Sharing environment variables are not supported, so we have to do it here again
+      - id: evaluate-inputs
+        name: Evaluate Inputs
+        run: |
+          branch=${BRANCH_NAME:-main}
```
Comment (Zelldon) [inline .github/workflows/qa-testbench.yaml:112]: Here the same

**Review by ChrisKujawa — COMMENTED** [review]
Thanks @oleschoenburg 

I have some questions, lets clarify them before we move forward, see below.


Q: I think we need to backport the changes by Nicolas to not overwrite flaky tests outputs right?
Q: Should we backport actions that are only targeting main, there is no need for them on stable right?

File: .github/ISSUE_TEMPLATE/bug_report.md (hunk 1 above)
Comment (oleschoenburg) [reply .github/ISSUE_TEMPLATE/bug_report.md:6]: They are not, but I just copied over the .github folder from main :sweat_smile: I can remove it of course but I think I'd prefer to just keep everything as close to main as possible

File: .github/workflows/deploy.yml (hunk 3 above)
Comment (oleschoenburg) [reply .github/workflows/deploy.yml:6]: Yeah, it'll just not run and do no harm.

File: .github/workflows/qa-testbench.yaml (hunk 4 above)
Comment (oleschoenburg) [reply .github/workflows/qa-testbench.yaml:12]: Scheduled workflows only run on the main branch so adding this to the stable/8.0 branch has no effect. If we want to run QA for other branches and versions we'd do this by adding a second schedule on the main branch.

File: .github/actions/setup-zeebe/action.yml (hunk 2 above)
Comment (oleschoenburg) [reply .github/actions/setup-zeebe/action.yml:19]: Ah, good idea to double check! Luckily 8.0 uses 17: https://github.com/camunda/zeebe/blob/stable/8.0/parent/pom.xml#L29

**Comment by oleschoenburg** [comment]
> Q: I think we need to backport the changes by Nicolas to not overwrite flaky tests outputs right?

Nice catch, forgot about that. I'll try and backport them now.

> Q: Should we backport actions that are only targeting main, there is no need for them on stable right?

Right, there's no need. But I'd prefer to backport as much as possible so that we don't have too many differences between stable and main. WDYT?

**Comment by Zelldon** [comment]
> Right, there's no need. But I'd prefer to backport as much as possible so that we don't have too many differences between stable and main. WDYT?

I was not sure. Because things that are not used I tend to delete since they might cause unnecessary confusion. I see no real value in backporting them if they aren't used.

**Comment by oleschoenburg** [comment]
> Q: I think we need to backport the changes by Nicolas to not overwrite flaky tests outputs right?

Hmm, how important is this? Backporting these changes is a major pain because of dependency convergence issues. Nicolas' changes require surefire 3.0.0-M7 but 8.0 is on 3.0.0-M5. Just backporting https://github.com/camunda/zeebe/pull/9049 too is not enough, the maven plugin versions have drifted quite far and there are other changes to our POMs. As we lived with flaky test output being overwritten for so long I'd lean towards not investing more time into backporting this.

**Comment by Zelldon** [comment]
Yeah maybe we can just disable the rerun :shrug:

**Comment by oleschoenburg** [comment]
> Yeah maybe we can just disable the rerun

How about a compromise: I can create a follow up issue to backport https://github.com/camunda/zeebe/pull/10520 and for now we leave this as is.

If that's okay with you, the PR should be ready to merge from my point of view.

**Review by ChrisKujawa — APPROVED** [review]
Thanks 👍
